app.js

```diff
@@ -564,14 +582,31 @@ Promise.all([findCompilers(), aws.initConfig(awsProps)])
 
         logger.info("=======================================");
         logger.info("Listening on http://" + (hostname || 'localhost') + ":" + port + "/");
-        logger.info("  serving static files from '" + staticDir + "'");
         if (gitReleaseName) logger.info("  git release " + gitReleaseName);
 
         function renderConfig(extra) {
             const options = _.extend(extra, clientOptionsHandler.get());
             options.compilerExplorerOptions = JSON.stringify(options);
             options.root = versionedRootPrefix;
             options.extraBodyClass = extraBodyClass;
+            options.builtResourcesRoot = builtResourcesRoot;
+            options.require = function(path) { 
+                if(process.env.NODE_ENV == "DEV") {
+                    //I have no idea why main => maps to styles i need to dig into this
+                    if(path == 'main.css') {
+                        return '/dist/styles.css';
+                    }
+                    //this will break assets in dev mode for now
+                    return '/dist/' + path;
+                }
+                if(staticManifest.hasOwnProperty(path)) {
+                    return versionedRootPrefix + "/dist/" + staticManifest[path];
+                }
+                if(assetManifest.hasOwnProperty(path)) {
+                    return versionedRootPrefix + "/dist/assets/" + assetManifest[path];
+                }
+                console.warn("requested an asset I don't know about");
```

Comment:
`logger.{warn,error}` better than console.log (Which will not let the tests pass)